src/Perspex.Controls/Presenters/TextPresenter.cs

```diff
@@ -93,7 +93,14 @@ public override void Render(IDrawingContext context)
             if (selectionStart == selectionEnd)
             {
                 var charPos = FormattedText.HitTestTextPosition(CaretIndex);
-                Brush caretBrush = Brushes.Black;
+
+                var backgroundColor = (((Control)TemplatedParent).GetValue(BackgroundProperty) as SolidColorBrush).Color;
```

Comment:
If the background isn't a SolidColorBrush, you're going to get a NRE here. I think you meant:

(((Control)TemplatedParent).GetValue(BackgroundProperty) as SolidColorBrush)?.Color;